app/services/bank_holiday_retriever.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class BankHolidayRetriever
+  UnsuccessfulRetrievalError = Class.new(StandardError)
+  API_URL = 'https://www.gov.uk/bank-holidays.json'
+  DEFAULT_GROUP = 'england-and-wales'
+
+  def self.dates
+    new.dates(DEFAULT_GROUP)
+  end
+
+  def data
+    return raise_error unless response.is_a?(Net::HTTPOK)
+
+    @data ||= JSON.parse(response.body)
+  end
+
+  def dates(group)
+    return if data.empty?
+
+    data.dig(group, 'events')&.map { |details| details['date'] }
```

Comment:
would `.pluck('date')` work instead of `.map { |details| details['date'] }` ?

Reply:
Yes - it might well do.